umbra-js/src/classes/Umbra.ts

```diff
@@ -151,7 +134,8 @@ export class Umbra {
    */
   constructor(readonly provider: EthersProvider, chainConfig: ChainConfig | number) {
     this.chainConfig = parseChainConfig(chainConfig);
-    this.umbraContract = new Contract(this.chainConfig.umbraAddress, umbraAbi, provider) as UmbraContract;
+    this.umbraContract = new Contract(this.chainConfig.umbraAddress, UMBRA_ABI, provider) as UmbraContract;
+    this.batchSendContract = new Contract(this.chainConfig.batchSendAddress, UMBRA_BATCH_SEND_ABI, provider);
```

Comment:
Similarly, this should be not initialized/undefined for chains that don't have the contract deployed

Reply:
Initialize only if `batchSendAddress` exists on `chainConfig` https://github.com/ScopeLift/umbra-protocol/pull/426/commits/c6ae57a55c8671619e38b1dd202e691fe4730098

Reply:
It looks like our `batchSend` method now just silently becomes a no-op that doesn't actually send funds if the batch send contract doesn't exist? That feels like confusing behavior. We should instead throw an error if you try to use `batchSend` on a chain that doesn't have it
<img width="668" alt="image" src="https://user-images.githubusercontent.com/17163988/232559590-773b9a47-a55f-4cde-baf1-ba20a153be0f.png">

Reply:
Similarly in the `parseChainConfig`, we don't need two whole if blocks like this:

<img width="463" alt="image" src="https://user-images.githubusercontent.com/17163988/232560032-5875c2e7-c1dc-441b-ae85-c325309ffd38.png">


Just do this:

```typescript
return {
      umbraAddress: getAddress(umbraAddress),
      batchSendAddress: batchSendAddress ? getAddress(batchSendAddress) : null,
      startBlock,
      chainId,
      subgraphUrl,
    };
```

Reply:
Aww man, I did it in such a roundabout way... Thank you! Addressed here: https://github.com/ScopeLift/umbra-protocol/pull/426/commits/961a34a4c697c5ca6dfce7c231990a6c95e2547f